### PR TITLE
Fix/refactor fetching

### LINKED ISSUE #25: Refactor data fetching to a single utils file

### PR DESCRIPTION
**Pull request by willstenzel**
Refactored api fetches to use functions from an api.js file 

Closes #25

### CODE REVIEW

File: src/Common/SideBar/SideBar.js [hunk 1]
```diff
@@ -12,66 +11,31 @@ import {
...
-        }
+    updateStudentCourseList = async (method_type, course_name) => {
+        const {data, error} = await apiUpdateStudentCourseList(method_type, course_name)
+        this.setState({courses_list: data.courses, error: error})
     }
 
-    async componentDidMount(){
-        await this.fetch_courses_list();
-        this.setState({isTA: await AsyncStorage.getItem('isTA')})
+    async componentDidMount() {
```
Comment (ryandrew14) [inline src/Common/SideBar/SideBar.js]: Did a bit of research here. Last meeting I talked about I thought how promises might be better, like this:
```
componentDidMount(){
    apiFetchStudentCourseList().then((data, error) => {
        this.setState({courses_list: data.courses, student_name: data.full_name, fetch_error: error, loading: false})
    }
}
```
But looks like the commonly accepted thing is to just use `async`: https://dev.to/gafi/7-reasons-to-always-use-async-await-over-plain-promises-tutorial-4ej9

So, I really don't know what the best thing is here. To me, `async componentDidMount` is unusual and therefore confusing so I do like the promise notation, but at the same time if it gets more complicated we would probably go to async/await syntax so I honestly don't know what the move is. Up to you -- as long as it's consistent I think either is good

File: src/Student/HomeScreen/OfficeHoursSchedule.js [hunk 2]
```diff
@@ -10,65 +11,47 @@ class OfficeHoursSchedule extends React.Component {
...
-                        'Authorization': "Token a891e91d45001088b201b3c2ebe8a5e87a9121f9",
-                    }});
-
-                const officeHours = await apiCall.json();
-                this.setState({officeHours: officeHours, loading: false});
-                console.log('office hours', officeHours)
-            } catch (err) {
-                console.log("Error fetching office hours", err);
-            }    this.representHours();
+            const {data, error} = await apiFetchCourseSchedule(course_id);
```
Comment (ryandrew14) [inline src/Student/HomeScreen/OfficeHoursSchedule.js]: So much cleaner!

File: src/TA/HomeScreen/TAHomeScreen.js [hunk 3]
```diff
@@ -4,80 +4,53 @@ import moment from 'moment'
...
         super(props);
-        this.state = {isLoading: true}
+        this.state = {loading: true}
     }
 
-
     componentDidMount() {
-        console.log("rendering TA Home Screen")
-        this.fetchUpcomingOfficeHours()
+        apiFetchUpcomingOfficeHours().then(({data, error}) => {
```
Comment (ryandrew14) [inline src/TA/HomeScreen/TAHomeScreen.js]: We should be consistent with how we handle this -- see my other comment about promise vs. async/await

File: src/Common/SideBar/SideBar.js (hunk 1 above)
Comment (willstenzel) [reply src/Common/SideBar/SideBar.js]: Good info! I just checked out the article and I think the async await will be a better way to go given that we are already using it a good amount in the code. I just changed the two places where we used .then() to use await instead which should make everything consistent now.

**Pull request merged**